### PR TITLE
Updated DB architecture to remove sections

### PR DESCRIPTION
**Pull request by NickLewanowicz**
- [x] Feature
- [ ] Issue

## Related Issue/Task

This isnt related to a current issue/task

## Description

To simplify the db architecture I removed 'section' model from mongo. From the UI projects will still have a sections array but that will be task objects, these tasks then have a tasks array themselves. We still have a functional idea of 'sections' on the ui as anything who doesnt have a section as a parent. 

## Motivation and Context

This will simply the GQL layer by only having 'createProject' and 'createTask' where 'createTask' accepts either a 'section' as a parent or a 'project' as a parent. 

## How Has This Been Tested?

Manually tested on local, will need to write tests. 

## Screenshots (if appropriate):

### CODE REVIEW

File: db/db.js [hunk 1]
```diff
@@ -54,6 +53,11 @@ const deleteRecord = async (model, attrs) => {
 
 // Model DB Utils ------------------------------------------------------------------------------
 
+// TODO: See if there is a better way to do this. ie, make find refs accept an array of key/values
+const findParentTasks = async (id) => {
+    return await Task.where('project').equals(id).where('section').equals(null)
```
Comment (ermel272) [inline db/db.js]: Could we achieve this by maintaining a list of parent IDs in each task object? To find parent task IDs we would just return the list and if we want the documents themselves iterate over the IDs querying for the respective documents?

File: models/task.js [hunk 2]
```diff
@@ -6,9 +6,14 @@ const taskSchema = new mongoose.Schema({
     unique: false,
     default: ''
   },
+  project: { 
+    type: mongoose.Schema.Types.ObjectId, 
+    ref: 'Project',
+    index: true
+  },
   section: {
```
Comment (Foopert) [inline models/task.js]: If we're not using sections anymore in our schema, then we should remove this. A potential solution would be renaming this "section" to a "sub-task", or something similar (if we still need it).

Edit: I notice that we're using "section" quite a bit through our code. Are we still going to be using sections in our schema?

File: db/db.js (hunk 1 above)
Comment (NickLewanowicz) [reply db/db.js]: A task should only ever have 1 parent and 1 project pointer. The purpose of this is to be a helper function that will get all the direct children of a project. Previously the 'section' model had a project pointer and no section, now that all tasks have project attribute we only want the ones that have a null section attribbute

Comment (NickLewanowicz) [reply db/db.js]: I would definitely NOT want to maintain a list of ids anywhere for any reason other then a many to many relation.

File: models/task.js (hunk 2 above)
Comment (NickLewanowicz) [reply models/task.js]: We can still call the attribute 'section' without it being a section model. The task model has 2 parents, the root parent which is 'project' and immediate parent which is 'section'

Comment (NickLewanowicz) [reply models/task.js]: this is a parent relation not a child so sub-task doesnt make sense.

File: db/db.js (hunk 1 above)
Comment (ermel272) [reply db/db.js]: Right so this query makes more sense... We're search for tasks that share the same project id right? Only part I don't get is why the section needs to be null..

Comment (ermel272) [reply db/db.js]: Discussed offline - maintaining a list of IDs is bad. This solution assumes in the future that direct children of projects will never populate the 'section' field. A preferred solution would be maintain a parent pointer in each task, then query for tasks in which the parent is the project id.

**Pull request merged**